server/index.js

```diff
@@ -96,62 +153,43 @@ app.get('/api/reviews/:chefId', (req, res) => {
     .catch(err => console.error(err));
 });
 
-app.post('/api/auth/sign-up', (req, res, next) => {
-  const { username, password } = req.body;
-  if (!username || !password) {
-    throw new ClientError(400, 'username and password are required fields');
+app.get('/api/search/:cuisineType', (req, res) => {
+  const cuisineType = req.params.cuisineType;
+  if (Number.isInteger(cuisineType)) {
+    res.status(400).json({ error: 'cuisineType must be letters' });
   }
-  argon2
-    .hash(password)
-    .then(password => {
-      const sql = `
-      insert into "users" ("username", "password", "createdAt")
-      values ($1, $2, current_timestamp)
-      returning "username", "createdAt", "userId"
-      `;
-      const params = [username, password];
-      db.query(sql, params)
-        .then(result => {
-          const [newUser] = result.rows;
-          res.status(201).json(newUser);
-        })
-        .catch(err => next(err));
+  const sql = `
+    select   "chefId", "chefs"."name", "photoUrl", avg(distinct "rating"), count(distinct "reviewId")
+    from     "chefs"
+    join     "reviews" using ("chefId")
+    join     "chefCuisines" using ("chefId")
+    join     "cuisines" using ("cuisineId")
+    where    "cuisines"."name" = $1
+    group by "chefs"."chefId"
+  `;
+  const params = [cuisineType];
+  db.query(sql, params)
+    .then(result => {
+      const [cuisines] = result.rows;
+      if (!cuisines) {
+        res.json([]);
+      } else {
+        res.json(result.rows);
+      }
     })
-    .catch(err => next(err));
+    .catch(err => console.error(err));
```

Comment:
Always pass this to `next()` so your error middleware can deal with it and the client doesn't hang